apps/web/pages/api/basenames/frame/tx.ts

```diff
@@ -0,0 +1,120 @@
+import { NextApiRequest, NextApiResponse } from 'next/dist/shared/lib/utils';
+import {
+  FrameRequest,
+  getFrameMessage,
+  FrameTransactionResponse,
+} from '@coinbase/onchainkit/frame';
+import { encodeFunctionData, namehash } from 'viem';
+import { base } from 'viem/chains';
+import L2ResolverAbi from 'apps/web/src/abis/L2Resolver';
+import RegistrarControllerABI from 'apps/web/src/abis/RegistrarControllerABI';
+import { formatBaseEthDomain } from 'apps/web/src/utils/usernames';
+import {
+  USERNAME_L2_RESOLVER_ADDRESSES,
+  USERNAME_REGISTRAR_CONTROLLER_ADDRESSES,
+} from 'apps/web/src/addresses/usernames';
+
+export type TxFrameStateType = {
+  targetName: string;
+  formattedTargetName: string;
+  targetYears: number;
+  registrationPriceInWei: string;
+  registrationPriceInEth: string;
+};
+
+const RESOLVER_ADDRESS = USERNAME_L2_RESOLVER_ADDRESSES[base.id];
+const REGISTRAR_CONTROLLER_ADDRESS = USERNAME_REGISTRAR_CONTROLLER_ADDRESSES[base.id];
+const NEYNAR_API_KEY = process.env.NEXT_PUBLIC_NEYNAR_API_KEY;
```

Comment:
```suggestion
const NEYNAR_API_KEY = process.env.NEXT_PUBLIC_NEYNAR_API_KEY;

if (!NEYNAR_API_KEY) {
  throw new Error('missing NEYNAR_API_KEY')
}
```

Reply:
this will throw an error at build time if the api key is missing

Reply:
awesome, added